socorro/processor/processor_app.py

```diff
@@ -293,48 +177,56 @@ def process_crash(self, crash_id, ruleset_name, tmpdir):
         except CrashIDNotFound:
             # If the crash isn't found, we just reject it--no need to capture
             # errors here
-            self.processor.reject_raw_crash(
+            self.pipeline.reject_raw_crash(
                 crash_id, "crash cannot be found in raw crash storage"
             )
             return
         except Exception as exc:
             sentry_sdk.capture_exception(exc)
             self.logger.exception("error: crash id %s: %r", crash_id, exc)
-            self.processor.reject_raw_crash(crash_id, f"error in loading: {exc}")
+            self.pipeline.reject_raw_crash(crash_id, f"error in loading: {exc}")
             return
 
         # Fetch processed crash data--there won't be any if this crash hasn't
         # been processed, yet
         try:
-            processed_crash = self.source.get_processed(crash_id)
+            processed_crash = self.source.get_processed_crash(crash_id)
             new_crash = False
         except CrashIDNotFound:
             new_crash = True
             processed_crash = {}
 
         # Process the crash to generate a processed crash
         self.logger.debug("processing %s", crash_id)
-        processed_crash = self.processor.process_crash(
+        processed_crash = self.pipeline.process_crash(
             ruleset_name=ruleset_name,
             raw_crash=raw_crash,
             dumps=dumps,
             processed_crash=processed_crash,
             tmpdir=tmpdir,
         )
 
-        # Save data to crash storage
-        try:
-            self.logger.debug("saving %s", crash_id)
-            self.destination.save_processed_crash(raw_crash, processed_crash)
-            self.logger.info("saved %s", crash_id)
-        except PolyStorageError as poly_storage_error:
-            # Capture and log the exceptions raised by storage backends
-            for storage_error in poly_storage_error:
-                sentry_sdk.capture_exception(storage_error)
-                self.logger.error("error: crash id %s: %r", crash_id, storage_error)
-
-            # Re-raise the original exception with the correct traceback
-            raise
+        # Save data to crash storage destinations
+        self.logger.debug("saving %s", crash_id)
+        for dest in self.destinations:
```

Comment:
Previously, we had poly-storage and metrics counters and benchmarking wrappers. All that code goes away with this for loop here where it supports multiple destinations and handles the metrics emitting on its own.